src/ui/public/route_based_notifier/index.js

```diff
@@ -0,0 +1,47 @@
+import { includes, mapValues } from 'lodash';
+import Notifier from 'ui/notify/notifier';
+
+/*
+ * Caches notification attempts so each one is only actually sent to the
+ * notifier service once per route.
+ */
+export default function routeBasedNotifier($rootScope) {
```

Comment:
I don't know if we have a consistent convention around this but I've been naming the default function being exported the same as the variable name on the import side. So in this case I'd name this `RouteBasedNotifierProvider`. But again, maybe there's no real convention here, in which case what you have is fine too.